### ISSUE TITLE
IndexOutOfBoundException when creating codec for a case class with exactly 32 fields

### ISSUE THREAD

**Issue by lociam** [plokhotnyuk/jsoniter-scala#389]
Whenever you try to create a codec for a class with exactly 32 fields you get an IndexOutOfBoundException in the JsonCodecMaker:774.

I am defining the codec as 

```JsonCodecMaker.make[Product](CodecMakerConfig())```

And the thrown exception is:

```
Error:(24, 33) exception during macro expansion: 
java.lang.IndexOutOfBoundsException: 1
	at scala.collection.LinearSeqOptimized.apply(LinearSeqOptimized.scala:63)
	at scala.collection.LinearSeqOptimized.apply$(LinearSeqOptimized.scala:61)
	at scala.collection.immutable.Stream.apply(Stream.scala:200)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.$anonfun$make$51(JsonCodecMaker.scala:774)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
	at scala.collection.Iterator.foreach(Iterator.scala:937)
	at scala.collection.Iterator.foreach$(Iterator.scala:937)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.map(TraversableLike.scala:233)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.genReadNonAbstractScalaClass$1(JsonCodecMaker.scala:773)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.$anonfun$make$93(JsonCodecMaker.scala:1135)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.$anonfun$make$34(JsonCodecMaker.scala:702)
	at scala.collection.mutable.MapLike.getOrElseUpdate(MapLike.scala:206)
	at scala.collection.mutable.MapLike.getOrElseUpdate$(MapLike.scala:203)
	at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:80)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.withDecoderFor$1(JsonCodecMaker.scala:702)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.genReadVal$1(JsonCodecMaker.scala:1134)
	at com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker$Impl$.make(JsonCodecMaker.scala:1342)
    JsonCodecMaker.make[Product](CodecMakerConfig())
```

Looking at the source code, it looks like the problem lies in ```classInfo.fields.grouped(32).toSeq.map(_.zipWithIndex.foldLeft(0)```

Adding an extra field, to get 33 fields solves the problem.

**Comment by plokhotnyuk** [comment]
Hi, @lociam !

Please check which version of the jsoniter-scala do you have now?

It seems that the bug was fixed in v0.47.0: https://github.com/plokhotnyuk/jsoniter-scala/releases/tag/v0.47.0

**Comment by plokhotnyuk** [comment]
@lociam I'm closing it as duplicate... Please feel free to reopen it if that is not the same case.
